### ISSUE TITLE
Line numbers not read out while reading the code in the editor

### ISSUE THREAD

**Issue by AccessibilityTestingTeam-TCS** [microsoft/vscode#52429]
**Environment Details:** 
VSCode Version : 1.24.0
OS Version : Win10
Mac OS Version:HighSierra10.13.5 

**Additional Details:**
MAS Violated: MAS1.3.1
ScreenReader Version: NVDA2018.2, Jaws Version: 18.0.5038

**Repro Steps:**
1. Launch VS Code while NVDA/Jaws/VoiceOver is ON.
2. Open any file in the editor and press down arrow or up arrow key.

**Actual:**
Jaws/NVDA/VoiceOver does not read the line number while reading the code in the editor.

**Expected:**
ScreenReader should read the line number while reading the code in the editor so that user could know on which line he is editing the code.

**Recommendations:**
Refer below link which is repository of bug fixes code snippets:
https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx

**User Impact:**
User will not be able to know on which line he is editing the code..

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

Attachment for Reference: 
[A11y VSCode Editor Jaws not reading line no..pptx](https://github.com/Microsoft/vscode/files/2118774/A11y.VSCode.Editor.Jaws.not.reading.line.no.pptx)

Does this issue occur when all extensions are disabled?: Yes

**Comment by isidorn** [comment]
Lines are also not read out by Visual Studio or any other IDE / text editor I tried
@AccessibilityTestingTeam-TCS which IDE / text editors do this? And why would knowing the line number be helpful and not spam?

**Comment by AccessibilityTestingTeam-TCS** [comment]
@isidorn From accessibility point of view, every content that are visually appearing on screen should also be accessible to disabled user through screen reader and we are able to see line numbers. Secondly, I verified the same on Visual studio text editor, there screen reader is reading the line numbers.

**Comment by isidorn** [comment]
@AccessibilityTestingTeam-TCS Thanks for clarifying. Leaving up to @alexandrudima to decide, if this is standard practice then I would suggest to read out line number if they are visible - if a user finds this spammy he could always disable it via `"editor.lineNumber": "off"`
Feels low hanging, assigning to july so we decide what to do here

**Comment by dranov** [comment]
Related discussion: https://github.com/Microsoft/monaco-editor/issues/410

**Comment by alexandrudima** [comment]
I've discussed a while ago with a couple of folks in the Visual Studio team that implemented this, and they managed to do it by manipulating low level accessibility events sent out by Visual Studio and they've described it as a hack.

In other words, sadly, we have no mechanism in the w3c spec to implement this.

**Comment by Neurrone** [comment]
Reading the line number every time you move to it is extremely spammy. If this is implemented, it should only do so if the display of line numbers is enabled.

**Comment by zersiax** [comment]
I think we are missing a happy medium solution here.
If we look at the use cases for wanting to know line numbers, we see that you generally don't care about them until you do. You don't visually check every line's number, you check when you need to know the line number and ignore it for the rest.
In screen reader scenarios you can't do that, it's either on all the time or off all the time. The visual studio 2017 editor has them all the time even if reporting line numbers is off, unless that was fixed recently, which isn't really what you want either.
The way I've seen other editors do this , which I actually like, is to populate the "go to line" dialog with the current line number when it's invoked. That way, you can if you need to know the line number invoke, quickly check and dismiss.
The only other non-spammy solution I can think of is to add a dedicated shortcut key to query the currently focused line number. @alexandrudima  is this feasible?

**Comment by isidorn** [comment]
@zersiax awesome idea. I actually would also like that the go to line dialog contains the current line since my use case is that I have line numbers always hidden, but still sometimes I want to know the line.
@alexandrudima do you mind if I do a PR that would change this?

Code pointer https://github.com/Microsoft/vscode/blob/a684fe7ee136f89d92fa25ee0b8f9bdeacd104b6/src/vs/workbench/parts/quickopen/browser/gotoLineHandler.ts#L96

**Comment by alexandrudima** [comment]
@zersiax That is a great idea!

@isidorn Sure, PR most welcome. The author of the code is @bpasero. There's a duplicate implementation of that in the standalone editor if you wanna take a stab at it.

**Comment by bpasero** [comment]
Sounds good to me.

**Comment by isidorn** [comment]
Ok, created this item for me and assigned it to this milestone https://github.com/Microsoft/vscode/issues/65219

**Comment by isidorn** [comment]
Closing as duplicate https://github.com/microsoft/monaco-editor/issues/410
And we have done a workaroung for this on the vscode side.
To find out a line number a user should simply Cltr/cmd + G or F1 > Go to Line
